### PR TITLE
SearchJobPosts:BE

### PR DESCRIPTION
**Pull request by GabisPapin**
-created a function to get all job posts with categories and skills;
created a function which get job posts, skills and category by userId.

### CODE REVIEW

File: src/modules/jobPost/jobPost.controller.ts [hunk 1]
```diff
@@ -16,6 +16,16 @@ export class JobPostController {
     return this.jobPostService.getJobPostByUser(Number(userId));
   }
 
+  @Get()
+  getAllJobPost() {
+    return this.jobPostService.getAllJobPost();
+  }
+
+  @Get('skill/:id')
+  findJobPostByUserId(@Param('id') userId: number) {
```
Comment (lyudad) [inline src/modules/jobPost/jobPost.controller.ts]: return types?

File: src/modules/jobPost/jobPost.service.ts [hunk 2]
```diff
@@ -40,6 +40,21 @@ export class JobPostService {
     return jobByUser;
   }
 
+  async getAllJobPost() {
+    const allPosts = await this.jobPostRepository.find({
```
Comment (lyudad) [inline src/modules/jobPost/jobPost.service.ts]: use query builder please

File: src/modules/jobPost/jobPost.service.ts [hunk 3]
```diff
@@ -40,6 +40,21 @@ export class JobPostService {
...
+      relations: ['jobSkills', 'jobCategory'],
+    });
+    return allPosts;
+  }
+
+  async findJobPostByUserId(userId: number) {
+    const getAllByUserId = await this.jobPostRepository.find({
```
Comment (lyudad) [inline src/modules/jobPost/jobPost.service.ts]: query builder

File: src/modules/jobPost/jobPost.controller.ts (hunk 1 above)
Comment (GabisPapin) [reply src/modules/jobPost/jobPost.controller.ts]: I do this to return the job posts by userId with the user skills and category. But I think I can use another function which have the path user/:id.

File: src/modules/jobPost/jobPost.service.ts (hunk 2 above)
Comment (GabisPapin) [reply src/modules/jobPost/jobPost.service.ts]: changed to query builder

File: src/modules/jobPost/jobPost.service.ts (hunk 3 above)
Comment (GabisPapin) [reply src/modules/jobPost/jobPost.service.ts]: I delete this function and use getJobPostByUser, and change that to query builder.

**Pull request merged**